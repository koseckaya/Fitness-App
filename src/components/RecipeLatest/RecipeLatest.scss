@import "src/styles/vars";


.latest__card {
    display: grid;
    grid-template-columns: 324px auto;
    margin-bottom: 0px;
    background:white;
    border: 1px solid rgb(233, 233, 239);
    border-radius: 15px;
    padding: 38px 30px;
    display: grid;
    column-gap: 30px;
    margin-bottom: 30px;
    &-title {
        font-weight: 500;
        font-size: 18px;
        line-height: 27px;
        letter-spacing: 0.04em;
        text-transform: uppercase;  
        margin-bottom: 15px;

        text-overflow: ellipsis;
        overflow: hidden;
        word-break: break-word;
        user-select: none;
        display: -webkit-box;
        -webkit-line-clamp: 3;
        -webkit-box-orient: vertical;
    }
    &-description{
        margin-bottom: 30px;
    }
}

@include for-max-740px() {
    .latest__card {
        gap: 0;
        a {
            img {
               width: 214px; 
            }
        }
    }
}

@include for-max-640px() {
    .latest__card {
        padding: 30px 12px;
        grid-template-columns: 237px auto;
        gap: 0;
        a {
            img {
               width: 200px; 
            }
        }
    }
    .latest__card-desc {
            font-size: 14px;
    }
    .latest__card-title {
        font-size: 16px;
    }

}       

@include for-max-568px() {
    .latest__card {
        display: flex;
        flex-direction: column;
        gap: 1rem;
        a {
            display: flex;
            align-items: center;
            justify-content: center;
            img {
                width: 75%;
                display: flex;
                border-radius: 1rem;
            }
        }
    }
}