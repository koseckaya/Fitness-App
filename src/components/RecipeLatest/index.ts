export { default as RecipeLatest } from './RecipeLatest';
export type { Props as RecipeLatestProps } from './RecipeLatest';
