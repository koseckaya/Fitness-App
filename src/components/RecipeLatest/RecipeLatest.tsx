import React, { Component } from "react";
import { Button } from "../../components/Button";

import {
    RecipeApi,
    RecipeContextType,
    RecipeItem,
} from "./../../data";
import "./RecipeLatest.scss";
import { Link } from "react-router-dom";
import { RecipeContext } from "../../App";
import { getRecipeId } from "./../../helpers/helpers";


export interface Props {}
interface IState {
    items: RecipeApi[];
    page: number;
    oneRecipe: RecipeItem[];
    hasMore: boolean;
    search: string;
    from: number;
    to: number;
    category: string;
}


export default class RecipeLatest extends Component<Props, IState> {
    state: IState = {
        items: [],
        page: 1,
        oneRecipe: [],
        hasMore: true,
        search: "",
        from: 0,
        to: 20,
        category: "alcohol-free",
    };
    static contextType = RecipeContext;

    render() {
        const { recipes: items } = this.context as RecipeContextType;
        if (Object.keys(items).length === 0) {
            return null;
        }
        const value6 = Object.values(items)[6];
        const latestRecept = value6.recipe;
        const latestReceptUrl = getRecipeId(value6.recipe.shareAs);

        return (
            <div className="latest">
                <h3 className="recipes__title">Latest Recipe</h3>
                <div className="latest__card">
                    <Link to={"/recipes/" + latestReceptUrl}>
                        <img
                            className="latest__card-img"
                            src={latestRecept.image}
                            alt={latestRecept.label}
                        />
                    </Link>
                    <div className="latest__card-desc">
                        <div className="latest__card-title">
                            {latestRecept.label}
                        </div>
                        <div className="latest__card-description">
                            {latestRecept.ingredientLines?.map((item) => (
                                <div className="latest__card-text">{item}</div>
                            ))}
                        </div>
                        <Link to={"/recipes/" + latestReceptUrl}>
                            <Button
                                content="View Recipe"
                                type={Button.TYPES.PRIMARY}
                            />
                        </Link>
                    </div>
                </div>
            </div>
        );
    }
}
