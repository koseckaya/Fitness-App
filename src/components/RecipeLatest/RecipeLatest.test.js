import React from 'react';
import renderer from "react-test-renderer";
import RecipeLatest from './RecipeLatest';
//import { RecipeContext } from "../../App";

//const addItem = jest.fn()

describe("RecipeLatest component", () => {
    it('should render component with default props', () => {
        const tree = renderer
            .create(
                <RecipeLatest />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
})