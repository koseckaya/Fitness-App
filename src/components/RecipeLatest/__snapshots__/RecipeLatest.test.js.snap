// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeLatest component should render component with default props 1`] = `<div />`;
